### DISCUSSION TITLE
How to bypass school network restrictions

### DISCUSSION THREAD

**Discussion by GrinchERR** [3kh0/ext-remover#475] [Fixes JOIN TITANIUM NETWORK TO SHARE PLEASE]
How can I bypass my school's network restrictions for free? Because I can't install extensions, I can't get a free VPN. I would ask if you guys could make one, but that would be expensive. There was one in the past but it got shut down.  Is there a way to use a DNS as an unblocker? We aren't allowed to have our phones at all during the day, so I can't connect to my hotspot. Plz help or make a way to install extensions.

**Comment by 1vers1on** [comment]
use a proxy. https://unblocker.hoosiertransfer.repl.co/ or schooltutor.xyz

**Reply by PSYclownYT** [reply]
rammerhead links:

(https://googlesclassroom.ga/
](https://googlesclassroom.gq/)
](https://googlesclassroom.cf/
]()https://direct2.rammerhead.org/

Interstellar links: (allow popups to hide from history)
](https://photography.aksnapshots.com/
)](interstellarontop.fr.to)
](goarmy.us.to)

**Reply by PSYclownYT** [reply]
whenever you want to find any byp@sses, in the words bypass, proxy, exploit, sh1mmer, use @ for a, use 0 for o, and ! for i when messaging or opening tickets to prevent blockers and extensions from beaming your activity to the office.
try using LTmeat to suspend blocker

**Comment by amelia-bartlett** [comment]
proxy, vpn, tor etc
if you cant  install extensions, then maybe try sh1mmer?

**Reply by GrinchERR** [reply]
Im not doing shimmer and need a free vpn option.

**Reply by amelia-bartlett** [reply]
tor works pretty well if you want to unblock sites (if you can install android apps on a chromebook then you can get "orbot" from the google play store, it uses the VPNService API or can act as a standalone SOCKS5 proxy)

if you want a free VPN id personally _suggest_ ProtonVPN but you might find that another service suits your needs better.

As for proxies: Fog Network develops some pretty good in-browser proxies (as web apps), if you want HTTP(S)/FTP(S)/SOCKS4/SOCKS5 proxies then idk.  I'd recommend a VPN if you can afford the overhead from the encryption.

You could also consider selfhosting your own VPN.  There is a program called pivpn (originally designed for operation from a raspberry pi, but you can install it on any Linux machine, even x86 ones) that makes managing configuration a lot easier.  Only thing is it isn't compatible with the builtin chrome os client in openvpn mode (even if you use an openvpn to onc converter, the chromeos client doesn't support some of the crypto settings it uses, i haven't tested the new wireguard client yet since pivpn tends to screw up its firewall config and become unreachable in wireguard mode)
You should check with your ISP whether they use CGNAT if you experience difficulty port forwarding - also use an external port testing service (make sure it uses the right protocol as most test the TCP port, whereas OpenVPN runs on UDP by default) if you cannot access the service by its public IP/DNS on the same LAN as your router may not support haripin NAT - the port may still be accessible from the Internet.  Also remember that everything related to the VPN has to stay powered on!  This includes your router, the computer running the server, and any network hardware in between (the client does not have to remain powered).

Also, given you cannot install extensions, idk how you want to set up a vpn connection without the builtin client (which will likely not support most vpn server configurations, even when you can get raw config files for example protonvpn offers downloadable .ovpn configs for their servers).  The main reason that nobody will want to set up a VPN for you is not due to cost, but due to legal liability e.g. if you commited copyright infringement then the cops would come to the door of the guy hosting the VPN first.  I already have a VPN server set up (vps with fasthosts running PiVPN) but i won't give you access for this reason, even though I could afford the bandwith and processor power, especially if you are in a different timezone to me (i overprovisioned the vps).  I just cannot afford the legal liability (especially if the [Online (un)Safety Bill](https://action.openrightsgroup.org/stop-state-censorship-online-speech) gets passed (I live in the UK)).

You should really consider selfhosting a VPN, it isn't actually _that_ difficult.  I would try accessing whatever you selfhost from outside of both your home and school network first to rule out lack of hairpin nat and port filters respectively if it doesn't work.  (replace home network with the network you are hosting from if you are hosting from a different network).  Also try from a non-school device e.g. a mobile phone ([decent android openvpn client here](https://f-droid.org/packages/de.blinkt.openvpn/))

**Reply by GrinchERR** [reply]
Would the law still apply if to were somebody in the USA????? i know it would be slow tho.

**Comment by amelia-bartlett** [comment]
> How can I bypass my school's network restrictions for free? Because I can't install extensions, I can't get a free VPN. I would ask if you guys could make one, but that would be expensive. There was one in the past but it got shut down. Is there a way to use a DNS as an unblocker? We aren't allowed to have our phones at all during the day, so I can't connect to my hotspot. Plz help or make a way to install extensions.

ive done some testing of a new piece of software

if you can get a linux or unix environment on a machine (crostini probably wont be suitable, if you are trying on a chromebook that is not enrolled you will want mrchromebox altfw + a bootable linux usb) then you can use it

its called stunnel - you can use it to create a ssh tunnel, but you dont need root on the remote machine, its (supposedly) faster than a normal ssh tunnel, and you dont need to forward every port or manually configure SOCKSv5 proxies

basically it works kinda like a VPN - I use it since my school uses DPI to disrupt OpenVPN but as long as SSH is on port 443 (sslh demultiplexer may be useful if you have a web server as well) they don't care
